### PR TITLE
Four progressive commits to the new `test_tool_meister.py` module

### PR DESCRIPTION
**Pull request by portante**

### CODE REVIEW

File: lib/pbench/test/unit/agent/test_tool_meister.py [hunk 1]
```diff
@@ -15,241 +15,188 @@
...
+            assert directory == tmp_dir
+            self.functions_called.append("fake_rmtree")
 
-    @responses.activate
-    def test_tar_create_success(self, monkeypatch, logger_fixture):
-        """This test should pass the tar creation in send directory"""
+        def fake_md5(tar_file: pathlib.Path):
+            assert tar_file == pathlib.Path(f"{self.directory}.tar.xz")
+            self.functions_called.append("fake_md5")
+            return 10, "random_md5"
```
Comment (npalaska) [inline lib/pbench/test/unit/agent/test_tool_meister.py:161]: Right, making this local to this test function make sense, no other tests are using this mocks and it keeps the unnecessary complexity out.

File: lib/pbench/test/unit/agent/test_tool_meister.py [hunk 2]
```diff
@@ -15,241 +15,188 @@
...
+    "group": "",
+    "hostname": "test.hostname.com",
+    "label": None,
+    "tool_metadata": {"persistent": {}, "transient": {}},
+    "tools": [],
+}
+
+
+@pytest.fixture
+def tool_meister():
```
Comment (npalaska) [inline lib/pbench/test/unit/agent/test_tool_meister.py:33]: Ah! that's clever.

File: lib/pbench/test/unit/agent/test_tool_meister.py [hunk 3]
```diff
@@ -15,241 +15,188 @@
...
-
-    log_level = marker.args[0] if marker else logging.ERROR
-    logger.setLevel(log_level)
-    yield logger
-    logger.setLevel(level=logging.NOTSET)
+class TestCreateTar:
+    """Test the ToolMeister._create_tar() method behaviors."""
 
+    @staticmethod
+    def test_create_tar(tool_meister, monkeypatch):
```
Comment (npalaska) [inline lib/pbench/test/unit/agent/test_tool_meister.py:49]: Why `staticmethod` though?

**Review by npalaska — APPROVED** [review]

File: lib/pbench/test/unit/agent/test_tool_meister.py (hunk 3 above)
Comment (portante) [reply lib/pbench/test/unit/agent/test_tool_meister.py:49]: There is no need for a `self` parameter for these tests.

**Review by dbutenhof — APPROVED** [review]
I'm OK with this; I'm a bit concerned about dropping the Logger hackery, because the genesis of that was trying to keep Pytest from dumping the `WARNING` messages on the console during successful retry tests. This "meta-PR" didn't run tests, so if that's still a problem we won't see it until this gets merged.

**Comment by portante** [comment]
> I'm OK with this; I'm a bit concerned about dropping the Logger hackery, because the genesis of that was trying to keep Pytest from dumping the `WARNING` messages on the console during successful retry tests. This "meta-PR" didn't run tests, so if that's still a problem we won't see it until this gets merged.

Not sure how the logs were being seen, but I don't see any warnings in the output from running successful tests.

**Pull request merged**